pandas/tests/series/indexing/test_indexing.py

```diff
@@ -444,3 +445,61 @@ def test_setitem_dict_and_set_disallowed_multiindex(self, key):
         ser = Series([1, 2], index=MultiIndex.from_tuples([(1, 2), (3, 4)]))
         with pytest.raises(TypeError, match="as an indexer is not supported"):
             ser.loc[key] = 1
+
+
+class TestSetitemValidation:
+    # This is adapted from pandas/tests/arrays/masked/test_indexing.py
+    # but checks for warnings instead of errors.
+    def _check_setitem_invalid(self, ser, invalid, indexer):
+        msg = "Setting an item of incompatible dtype is deprecated"
+        msg = re.escape(msg)
+
+        orig_ser = ser.copy()
+
+        with tm.assert_produces_warning(FutureWarning, match=msg):
+            ser[indexer] = invalid
+            ser = orig_ser.copy()
+
+        with tm.assert_produces_warning(FutureWarning, match=msg):
+            ser.iloc[indexer] = invalid
+            ser = orig_ser.copy()
+
+        with tm.assert_produces_warning(FutureWarning, match=msg):
+            ser.loc[indexer] = invalid
+            ser = orig_ser.copy()
+
+        # note: commented-out in the EA case too
+        # FIXME: don't leave commented-out
+        # with tm.assert_produces_warning(FutureWarning, match=msg):
+        #     ser[:] = invalid
```

Comment:
this doesnt match the commented-out code in the masked tests.  any chance it can be un-commented?